packages/webcert/src/feature/careProvider/CareProviderModal.test.tsx

```diff
@@ -0,0 +1,204 @@
+import React from 'react'
+import { render, screen } from '@testing-library/react'
+import { Provider } from 'react-redux'
+import { BrowserRouter } from 'react-router-dom'
+import { updateUser, updateUserStatistics } from '../../store/user/userActions'
+import { LoginMethod, SigningMethod, Unit, User, UserStatistics, UnitStatistic, UnitStatistics } from '@frontend/common'
+import CareProviderModal from './CareProviderModal'
+import userEvent from '@testing-library/user-event'
+import { configureStore, EnhancedStore } from '@reduxjs/toolkit'
+import reducer from '../../store/reducers'
+import apiMiddleware from '../../store/api/apiMiddleware'
+import { userMiddleware } from '../../store/user/userMiddleware'
+import MockAdapter from 'axios-mock-adapter'
+import axios from 'axios'
+import dispatchHelperMiddleware, { clearDispatchedActions, dispatchedActions } from '../../store/test/dispatchHelperMiddleware'
+
+let fakeAxios: MockAdapter
+let testStore: EnhancedStore
+
+// https://stackoverflow.com/questions/53009324/how-to-wait-for-request-to-be-finished-with-axios-mock-adapter-like-its-possibl
+const flushPromises = () => new Promise((resolve) => setTimeout(resolve))
+
+const renderComponent = () => {
+  render(
+    <Provider store={testStore}>
+      <BrowserRouter>
+        <CareProviderModal />
+      </BrowserRouter>
+    </Provider>
+  )
+}
+
+describe('Care provider modal', () => {
+  beforeEach(() => {
+    fakeAxios = new MockAdapter(axios)
+    testStore = configureStore({
+      reducer,
+      middleware: (getDefaultMiddleware) => getDefaultMiddleware().prepend(dispatchHelperMiddleware, apiMiddleware, userMiddleware),
+    })
+  })
+
+  afterEach(() => clearDispatchedActions())
+
+  it('should show care provider modal if logged in unit is not set', () => {
+    testStore.dispatch(updateUser(getUserWithEmptyUnit()))
+    testStore.dispatch(updateUserStatistics(getUserStatistics()))
+
+    renderComponent()
+    expect(screen.queryByRole('dialog')).toBeInTheDocument()
+  })
+
+  it('should not show modal if logged in unit is set', () => {
+    testStore.dispatch(updateUser(getUser()))
+
+    renderComponent()
+    expect(screen.queryByRole('dialog')).not.toBeInTheDocument()
+  })
+
+  it('should close modal when unit is chosen', async () => {
+    testStore.dispatch(updateUser(getUserWithEmptyUnit()))
+    testStore.dispatch(updateUserStatistics(getUserStatistics()))
+    fakeAxios.onPost('/api/user/unit/1234a').reply(200, { user: getUser() })
```

Comment:
Because we wanted to test the click on a unit in the modal and we needed the onPost to update the store with new data. We could have updated the store directly but then the click would have been unecessary.